autowiring/auto_signal.h

```diff
@@ -27,16 +27,33 @@ namespace autowiring {
         pBlink(nullptr)
       {}
 
-      virtual ~signal_node_base(void) {}
+      /// <summary>
+      /// Removes this node from the list it's in, or throws if it's the head.
+      /// </summary>
+      virtual ~signal_node_base(void) {
+       // Clear linkage
```

Comment:
Recommend you do not unlink from the destructor.  There are no guarantees afforded w.r.t. synchronization here, and you cannot add them after-the-fact if you decide you need them later.